openapi/composer-koji/openapi.yml

```diff
@@ -0,0 +1,145 @@
+---
+openapi: 3.0.1
+info:
+  version: '1'
+  title: OSBuild Composer - Koji
+  description: Service to build and push images to Koji.
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+
+paths:
+  /openapi.json:
+    get:
+      summary: Get the OpenAPI JSON specification
+      tags:
+        - meta
+      responses:
+        '200':
+          description: Returns this document
+
+  /compose/{compose_id}:
+    get:
+      summary: The status of a compose
+      parameters:
+        - in: path
+          name: compose_id
+          schema:
+            type: string
+            format: uuid
+            example: '123e4567-e89b-12d3-a456-426655440000'
+          required: true
+          description: ID of compose status to get
+      description: Get the status of a running or finished compose. This includes whether or not it succeeded, and also meta information about the result.
+      operationId: compose_status
+      responses:
+        '200':
+          description: compose status
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ComposeStatus'
+
+  /compose:
+    post:
+      summary: Create compose
+      description: Create a new compose, potentially consisting of several images and upload each to koji.
+      operationId: compose
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ComposeRequest'
+      responses:
+        '200':
+          description: Compose has started
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ComposeResponse'
+
+components:
+  schemas:
+    ComposeStatus:
+      required:
+        - status
+        - koji_task_id
+      properties:
+        status:
+          type: string
+          enum: ['success', 'failure', 'pending', 'registering']
+          example: 'success'
+        image_statuses:
+          type: array
+          items:
+            $ref: '#/components/schemas/ImageStatus'
+        koji_task_id:
+          type: integer
+          example: 203143
+    ImageStatus:
+      required:
+       - status
+      properties:
+        status:
+          type: string
+          enum: ['success', 'failure', 'pending', 'building', 'uploading']
+          example: 'success'
+    ComposeRequest:
+      type: object
+      required:
+        - distribution
+        - image_requests
```

Comment:
Shouldn't koji be also required? You know, every optional field makes the code more complex.

Reply:
Fixed.